### PR TITLE
feat: Show recent datasets in load dropdown

### LINKED ISSUE #122: Add recent datasets to load dropdown
- Show the last 4-5 loaded datasets in a list on the Load modal.
- Dataset/collection URLs would be saved via browser local storage.

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
Problem
=======
Closes #122!

Solution
========
- Clicking on the input field in the Load menu will show a list of the 10 most recently opened datasets, with the most recent datasets at the top.
- Successfully opened datasets are saved to local storage (using the new `usehooks-ts` library), so they persist between browser sessions.

Misc:
- Adds `assets/index.ts` to start isolating some weird import syntax to a single location, specifically for SVG icon used in `LabeledDropdown`.
- Caught some typos in documentation elsewhere in the project.
- Fixes a bug where `fetchWithTimeout` would not actually time out (was able to catch this due to today's Vast outage 😋)
  - Made a new issue for adding unit tests for this (#142)

## Type of change
* New feature (non-breaking change which adds functionality)

Change summary:
---------------
* Shows a list of the 10 most recently opened datasets in the Load modal, adding open/close behavior.
* Tweaks/improvements to Load modal UI.
* Adds the `usehooks-ts` library.
* Fixes issue with `fetchWithTimeout` where it would not time out.

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-141/
1. Go to the Load menu, and load in a dataset (https://dev-aics-dtp-001.int.allencell.org/dan-data/colorizer/data/test)
1. Open the load menu again. There should now be two options in the recent datasets dropdown.
1. Clicking an option in the dropdown will populate it in the input.
1. Open another dataset (https://dev-aics-dtp-001.int.allencell.org/dan-data/colorizer/EMT/collection.json).
1. Test switching between datasets and loading each one. The list should be ordered by most recent still.

Screenshots (optional):
-----------------------
image

image

🔉 (Turn on audio for walkthrough!)

https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/5cfcd484-e63f-4e26-8b83-a28baaf2fa6c

Keyfiles (delete if not relevant):
-----------------------
1. `LoadDatasetButton.tsx`

Thanks for contributing!

### CODE REVIEW

File: src/assets/index.ts
Comment (ShrimpCryptid) [inline src/assets/index.ts:1]: Kind of a throwaway change I added that affects `LabeledDropdown.tsx`. Helps to make the imports a little more organized!

File: src/components/LoadDatasetButton.tsx [hunk 1]
```diff
@@ -77,22 +233,48 @@ export default function LoadDatasetButton(props: LoadDatasetButtonProps): ReactE
...
+                  />
+                </Dropdown>
+                <Button type="primary" onClick={handleLoadClicked} loading={isLoading}>
+                  Load
+                </Button>
+              </Space.Compact>
+            </div>
+            <p>
+              <i>
+                <span style={{ color: theme.color.text.hint }}>Click for recent datasets</span>
```
Comment (interim17) [inline src/components/LoadDatasetButton.tsx:273]: Nit: I was clicking on this text expecting a reaction before I realized I should click on the input field, slightly confusing. Could replace example URL text, or a tooltip? Not a big deal.

**Review by interim17 — APPROVED** [review]

**Comment by interim17** [comment]
LGTM and works on my end.

File: src/components/LoadDatasetButton.tsx (hunk 1 above)
Comment (ShrimpCryptid) [reply src/components/LoadDatasetButton.tsx:273]: Ended up moving the text on top, and explicitly stating "Click below". Thank you!

Comment (ShrimpCryptid) [reply src/components/LoadDatasetButton.tsx]: image

**Comment by toloudis** [comment]
I feel like the code would be simpler if there were just a separate dropdown below the input field.

**Review by toloudis — APPROVED** [review]

**Review by toloudis — APPROVED** [review]
LGTM!

**Pull request merged**